module: system/make

LinusU opened the pull request:
This is a new module that runs the `make` command on the target. The benefit of using this module instead of the `shell` module is that it supports "check mode" and only reports `changed` where appropriate.

LinusU commented:
Code is now also 100% PEP8 compliant :+1:

robynbergeron commented:
Hi Linus,

Thanks for submitting this new module. ;)

Per the Ansible module guidelines: http://docs.ansible.com/developing_modules.html#module-checklist -- can you update the author line and add in your github ID? Thanks!

LinusU commented:
Sorry, I actually read thru that but I must have missed that line. Amending now.

LinusU commented:
Looking around for how I should specify it I noticed that there are some inconsistencies in how each module specify it's info. I changed the author line to be `Full name (@GithubName) <EmailAddress>`. I also changed the `version_added` to be a string; there are still a number of modules that have this as a float thought.

LinusU commented:
@robynbergeron Care to review my changes? :smile: :+1:

robynbergeron commented:
@LinusU yeah, there are definitely inconsistencies around :) Doing our best to catch up with the old ones, and make sure new things are coming in with the right info. The most useful thing for myself and @gregdek as triage folks is to be able to actually find a github ID quickly rather than consulting the googleverse :) (I can't comment on the string vs. float -- not sure if that's something @gregdek has been looking at or not in our triage-a-thon?)

Thanks for the update -- I'll put this in community_review, with the new community PR review process info :)

robynbergeron commented:
Thanks for submitting this new module to Ansible Extras! This module is now in community review, a process that is open to all Ansible users. In order for this module to be approved, it must gain the following votes:

“works_for_me”: If you have tested the module thoroughly, including testing of all of the module’s options, and if the module works for you, please add “works_for_me” in the comments.

“passes_guidelines”: If you have gone through the module guidelines and the module meets all of the requirements, please add “passes_guidelines” in the comments. Guidelines are available here: http://docs.ansible.com/developing_modules.html#module-checklist

“needs_revision”: If the module fails to work for you, or if it doesn’t meet guidelines, please add “needs_revision” in the comments with details about what needs to be fixed.

When a module has both “works_for_me” and “passes_guidelines” tags, we will promote the module for inclusion in Ansible Extras.  At this point, you will be expected to maintain the module by fixing bugs and evaluating pull requests in a timely manner. 

Thanks again for submitting your Ansible module!

conorsch commented:
Great addition! Goes nicely with the `patch` module. Regarding the new revision rules:
- works_for_me
- passes_guidelines

Cheers!

LinusU commented:
Thank you! \o/

robynbergeron commented:
@conorsch -- thanks for reviewing this for both guidelines and functionality!

@LinusU -- Congratulations! You are now the maintainer of a module in Extras. Thanks so much for your contribution! 

Marking this shiny new module for inclusion... one of the first to be moving in under the new policy for reviewing new extras modules. YAY, validation!

(Sorry, I'm excited :D)

cc @ansible/core team.

conorsch commented:
@robynbergeron The new guidelines are a huge improvement—thanks for advocating for more inclusive reviews!

LinusU commented:
Awesome!!

robynbergeron commented:
@conorsch -- I'm delighted to hear that :) Tell your friends! :D I'm super hopeful that people will remember to pass on the review love their modules receive -- or that those giving reviews will have receive them back in the future :)

abadger commented on system/make.py:
```diff
@@ -0,0 +1,109 @@
+        supports_check_mode=True,
+        argument_spec=dict(
+            target=dict(required=False, default=None),
+            params=dict(required=False, default=None),
```
Would it make sense for params to be a dict type or is it really any free form string?

abadger commented:
@LinusU hey, I was just going through tickets I'd commented on and saw that this one still had my question about the type of the params argument.  Any information from you?

LinusU commented:
Sorry, this has fallen thru the cracks.

I think it makes sense to let `params` be a dict type, I'll update the code.

mscherer commented on system/make.py:
```diff
@@ -0,0 +1,109 @@
+module: make
+short_description: Run targets in a Makefile
+requirements: []
+version_added: "1.9"
```
I think this also need to be updated

sivel commented:
```
============================================================================
make.py
============================================================================
ERROR: version_added should be 2.1. Currently 1.9
ERROR: No RETURN documentation provided
ERROR: ansible.module_utils.basic import not near call to main()
```

gregdek commented:
@LinusU A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

LinusU commented:
Yes, sorry for the delay on this, I'll update as soon as I can :)

LinusU commented:
'ready_for_review'

Updated to use dict :raised_hands:

sivel commented:
There are still 2 failing checks:

```
============================================================================
make.py
============================================================================
ERROR: No RETURN documentation provided
ERROR: ansible.module_utils.basic import not near call to main()
```

LinusU commented:
Does anyone know how the `RETURN` doc should look when not returning any values?

Also, is there any reasoning on "ansible.module_utils.basic import not near call to main()" somewhere? It breaks PEP8.

LinusU commented:
Should work now :+1:

LinusU commented:
Thanks :raised_hands:
